### PR TITLE
samples: wpan_serial: Add rf2xx transceiver overlay

### PR DESCRIPTION
**Pull request by nandojve**
Add Atmel rf2xx transceiver and update documentation to add details how build and flash the application. The intention is differentiate between SoC and transceivers to help user understand what they need to do to build and flash successfully.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

File: samples/net/wpan_serial/README.rst [hunk 1]
```diff
@@ -13,7 +13,8 @@ Requirements
 ************
 
 The sample assumes that 802.15.4 radio and USB controller are supported on
-a board.
+a board. You can pick, for instance, a transceiver like CC2520/RF2xx using
+overlays or using a SoC with build in radio like kw41z/nrf5/samr21.
```
Comment (dbkinder) [inline samples/net/wpan_serial/README.rst]: Let's change this last sentence to:
```
You can pick, for example, a transceiver such as a CC2520 or RF2xx using
overlays, or by using an SoC with a built-in radio,  such as a kw41z, nrf5,
or samr21.
```

File: samples/net/wpan_serial/README.rst [hunk 2]
```diff
@@ -26,8 +27,46 @@ Building and Running
...
+   To build the wpan_serial sample:
+
+   .. zephyr-app-commands::
+     :zephyr-app: samples/net/wpan_serial
+     :board: <board name>
+     :conf: "prj.conf [overlay-<RADIO>.conf]"
+     :goals: build
+     :compact:
+
+   This is an example how build for the Atmel SAM R21 Xplained Pro
```
Comment (dbkinder) [inline samples/net/wpan_serial/README.rst]: Change to:
```
   Here's how to build the sample for the Atmel SAM R21 Xplained Pro
   Development Kit. Note that for this SoC, you don't need to include
   ``overlay-rf2xx.conf``.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by pfalcon — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
+1 for docs, thanks!

File: samples/net/wpan_serial/README.rst [hunk 3]
```diff
@@ -26,8 +28,47 @@ Building and Running
...
+   .. zephyr-app-commands::
+     :zephyr-app: samples/net/wpan_serial
+     :board: atsamr21_xpro
+     :conf: "prj.conf"
+     :goals: build
+     :compact:
+
+   Flash your board
+   .. zephyr-app-commands::
+     :zephyr-app:
```
Comment (jukkar) [inline samples/net/wpan_serial/README.rst]: Documentation error seems to be related to this section, you probably need to fill these couple of empty values

**Review by jukkar — APPROVED** [review]

**Pull request merged**